### ISSUE TITLE
Gqlgen not compliant with graphql spec for Int type

### ISSUE THREAD

**Issue by roeest** [99designs/gqlgen#3214]
### What happened?
Gqlgen treats the `Int` graphql type as a golang 64 bit integer. The [graphql spec](https://spec.graphql.org/October2021/#sec-Int) states that the `Int` graphql type represents a 32 bit integer. This causes issues with clients that assume the underlying type of an integer is 32 bits either by overflowing them or by making them error out altogether.

### What did you expect?

Behavior that matches the graphql spec
1. Error when specifying an `Int` input that overflows 32bit integers
2. Truncate `Int` results that overflow or even better return an indicative error

### Minimal graphql.schema and models to reproduce
Any schema with an `Int` in an input or a regular type
### versions
 - `go run github.com/99designs/gqlgen version`? v0.17.49
 - `go version`? go version go1.22.4 darwin/arm64

**Comment by phughes-scwx** [comment]
@roeest does updating the `gqlgen.yaml` models section so that:

```yaml
models:
  ID:
    model:
      - github.com/99designs/gqlgen/graphql.ID
      - github.com/99designs/gqlgen/graphql.Int32
  Int:
    model:
      - github.com/99designs/gqlgen/graphql.Int32
```

Make the server spec-compliant, as you see it?

**Comment by phughes-scwx** [comment]
I am trying to imagine a PR to introduce this here, and what I come up with is one that changes the init behavior and documentation... Including documentation that would specifically point out using Int64 / BigInt scalars. I think existing schemas are SOL on this: using int32 across the resolver interface is pretty much the best / only way to ensure users take this into consideration.

Are there any updates that you think should / must be made to either the scalar (Int, Int32, Int64) implementations or the codegen / graphql package to ensure we error correctly? I'm imagining adding a [safecast](https://github.com/rung/go-safecast/blob/main/integer.go#L7-L12) step to Int32 [Marshal](https://github.com/99designs/gqlgen/blob/master/graphql/int.go#L58) / [Unmarshal](https://github.com/99designs/gqlgen/blob/master/graphql/int.go#L62)... but I feel like I'm missing something maybe.

**Comment by phughes-scwx** [comment]
Ah I think I see now: we should have the graphql `Int` map to `int32` directly in the resolver when generating, not `int`.

I think that, in the context of the suggested change to the default yaml setup, we can say that it's not really important to have the inputs map to `int32` (since a client that sends that data will receive an error), but it would be useful to have result values map to in32 so that we are certain to handle it in the resolver implementation.

This is a pretty drastic change: I think maybe introducing a configuration setting in the yaml to do this and then advocating for that to become the default as well may be the best route to take...

**Comment by roeest** [comment]
I think the current marshal/unmarshal methods would work correctly as is, it's more about making sure that the right ones are being used by the mappings in the configuration. I was playing around with a CI step that parsed the configuration and made sure that the mapping for `Int` and `Int64` exist and are correct (only allow mapping `Int` to `graphql.Int32` and vice-versa). I think the default behavior when not specifying a mapping for `Int` is to add a mapping to the 64 bit types, so that would obviously need to be changed. You'd probably need a flag here to allow people to use the old behavior so as to not break their existing apis.

Built in Int64 scalar support in the would be nice, similarly to other built in scalars (`UUID`, `Time` etc.)

**Comment by phughes-scwx** [comment]
OK, this above PR actually solves the issue, but is pretty strict. I failed at my attempt to introduce a "partial compliance" mode to the config that allows `Int` to map to int/int64 for inputs but not for results. Will link to a partially completed version of that in a branch if anyone is interested in helping.

**Comment by phughes-scwx** [comment]
I think this can be closed given #3409. Anything else you want, @roeest?
